Libplanet/Action/ActionContext.cs

```diff
@@ -36,8 +42,21 @@ internal class ActionContext : IActionContext
 
         public IRandom Random { get; }
 
+        public HashDigest<SHA256>? PreviousStateRootHash =>
```

Comment:
How about moving this calculation to its constructor and turning `PreviousStateRootHash` into the read-only field?

Reply:
You mean using `readonly` keyword, not `{ get; }` property. Is it right?

Reply:
Nope, I think `{ get;  }` property is fine.

Reply:
The original intention was to make the calculation lazy (i.e., the `Commit` doesn't work every time). If it moved to its constructor, it will occur `Commit` every time. So, for the intention, I made a new field and cache it. Please see the commit, a174f6d.